website/content/gateway/api_reference/resources/authorizations/authorizations_methods/token.md

```diff
@@ -10,3 +10,93 @@ Two token frameworks are supported:
 
 * `token[m4m]`: Mastercard Digital Enablement Service (MDES) for Merchants (M4M)
 * `token[vts]`: Visa Token Service (VTS)
+
+{{% notice %}}
+**Notice**: Signing is required to use the `token` payment methods.
+{{% /notice %}}
+
+#### Method: token[m4m]
+
+The required values are found in either the MDES `transact` response or the Secure Card on File (SCOF) `checkout` response.
+
+{{% description_list %}}
+{{% description_term %}}token[m4m][tan] {{% regex %}}[0-9]{12,19}{{% /regex %}}{{% /description_term %}}
+{{% description_details %}}Token Account Number (TAN) to charge.
+
+Found in `encryptedPayload.encryptedData.accountNumber` (MDES) or `encryptedPayload.token.paymentToken` (SCOF).
+{{% /description_details %}}
+
+{{% description_term %}}token[m4m][expire_month] {{% regex %}}[0-9]{2}{{% /regex %}}{{% /description_term %}}
+{{% description_details %}}Expiry month of token to charge.
+
+Found in `encryptedPayload.encryptedData.applicationExpiryDate` (MDES) or `encryptedPayload.token.tokenExpirationMonth` (SCOF).
+{{% /description_details %}}
+
+{{% description_term %}}token[m4m][expire_year] {{% regex %}}20[0-9]{2}{{% /regex %}}{{% /description_term %}}
+{{% description_details %}}Expiry year of token to charge.
+
+Found in `encryptedPayload.encryptedData.applicationExpiryDate` (MDES) or `encryptedPayload.token.tokenExpirationYear` (SCOF).
+{{% /description_details %}}
+
+{{% description_term %}}token[m4m][eci] {{% regex %}}0[267]{{% /regex %}}{{% /description_term %}}
+{{% description_details %}}Electronic Commerce Indicator.
+
+Found in `eci` (SCOF).
```

Comment:
```suggestion
Found in `eci` (SCOF).
{{% regex_optional %}}Optional.{{% /regex_optional %}}
```
To some extend we could mark it the same way as `cryptogram`. But we have kept ECI optional even for CIT transaction, due to legacy, so let's make it optional without any limitations?
It needs to be optional for subsequent recurring at least. :)
I haven't observed any tokenized Mastercard transaction without ECI, so we could take the plunge and mark it the same way as for `cryptogram`: Required on CIT, otherwise optional.

Reply:
> It needs to be optional for subsequent recurring at least. :)

Well, I imagined client chosing actively to send ECI 07 for subsequent-in-series, as that's what they should be processed with. So for me it's not given that it has to be optional for subsequent-in-series.

It could also be optional with a default value; "Optional; defaults to 07."

I would like more inputs to this discussion :slightly_smiling_face:

Reply:
While we're on the subject of `token[m4m][eci]`: I've been unable to find `eci` in responses for [MDES](https://developer.mastercard.com/mdes-digital-enablement/documentation/api-reference/). I've only found it for SCOF. That's why the helping text for this parameter only mentions [SCOF](https://developer.mastercard.com/secure-card-on-file/documentation/new-api-reference/apis/).

Reply:
> Optional; defaults to 07.

I think that would work quite fine. So, we're talking about

```suggestion
Found in `eci` (SCOF).
{{% regex_optional %}}Required on CIT, otherwise optional (defaults to `07`).{{% /regex_optional %}}
```

right? (And perhaps the markup should become `regex_conditional` rather than `regex_optional` 😛)

Reply:
Right :+1: Let's go with that :+1:

Reply:
Sorry, I am late to noticing your reply. My point was that the PSP doesn't have a ECI to give on a subsequent recurring. Just like they don't have a TAV. Where would they get it from?
On initial in series they exchange a PAN to a TAN+TAV+ECI. On subsequent recurring the PSP way make a MIT using the TAN.

Reply:
I've pushed https://github.com/clearhaus/gateway-api-docs/pull/149/commits/93fc4b17d368fe72278163dc91d847fe364e4eb5. I'll resolve this thread. We can start a new one.